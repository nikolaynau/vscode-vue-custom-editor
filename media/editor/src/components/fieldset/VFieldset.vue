<script setup lang="ts">
import { computed, toRefs } from 'vue';

export interface Props {
  tag?: string;
  label?: string;
  tBorder?: boolean;
  bBorder?: boolean;
  padding?: boolean;
}

const props = withDefaults(defineProps<Props>(), {
  tag: 'div',
  label: undefined,
  tBorder: false,
  bBorder: false,
  padding: true
});

const { padding, tBorder, bBorder } = toRefs(props);

const cssClasses = computed(() => ({
  'v-fieldset--t-border': tBorder.value,
  'v-fieldset--b-border': bBorder.value,
  'v-fieldset--padding': padding.value
}));
</script>

<template>
  <component class="v-fieldset" :class="cssClasses" :is="tag">
    <div v-if="label" class="v-fieldset__label">{{ label }}</div>
    <slot></slot>
  </component>
</template>

<style>
@import 'fieldset';
</style>
