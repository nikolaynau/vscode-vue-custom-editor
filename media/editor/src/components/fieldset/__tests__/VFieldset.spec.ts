import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import VFieldset from '../VFieldset.vue';

describe('VFieldset', () => {
  it('default render props', () => {
    const wrapper = mount(VFieldset);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('label prop', () => {
    const wrapper = mount(VFieldset, { props: { label: 'Some label' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('default slot', () => {
    const wrapper = mount(VFieldset, { slots: { default: 'Text' } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('tag prop', () => {
    const wrapper = mount(VFieldset, { props: { tag: 'span' } });
    expect(wrapper.element.tagName).toBe('SPAN');
  });

  it('tBorder prop', () => {
    const wrapper = mount(VFieldset, { props: { tBorder: true } });
    expect(wrapper.classes('v-fieldset--t-border')).toBe(true);
  });

  it('bBorder prop', () => {
    const wrapper = mount(VFieldset, { props: { bBorder: true } });
    expect(wrapper.classes('v-fieldset--b-border')).toBe(true);
  });

  it('padding prop', () => {
    const wrapper = mount(VFieldset, { props: { padding: true } });
    expect(wrapper.classes('v-fieldset--padding')).toBe(true);
  });

  it('render label prop with text', () => {
    const wrapper = mount(VFieldset, {
      props: { label: 'Some label' },
      slots: { default: 'Text' }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
