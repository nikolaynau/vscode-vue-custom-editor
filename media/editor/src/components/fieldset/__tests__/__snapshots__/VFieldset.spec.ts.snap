// Vitest Snapshot v1

exports[`VFieldset > default render props 1`] = `
"<div class=\\"v-fieldset v-fieldset--padding\\">
  <!--v-if-->
</div>"
`;

exports[`VFieldset > default slot 1`] = `
"<div class=\\"v-fieldset v-fieldset--padding\\">
  <!--v-if-->Text
</div>"
`;

exports[`VFieldset > label prop 1`] = `
"<div class=\\"v-fieldset v-fieldset--padding\\">
  <div class=\\"v-fieldset__label\\">Some label</div>
</div>"
`;

exports[`VFieldset > label with text 1`] = `
"<div class=\\"v-fieldset v-fieldset--padding\\">
  <div class=\\"v-fieldset__label\\">Some label</div>Text
</div>"
`;

exports[`VFieldset > render label prop with text 1`] = `
"<div class=\\"v-fieldset v-fieldset--padding\\">
  <div class=\\"v-fieldset__label\\">Some label</div>Text
</div>"
`;
